pym/bob/input.py

```diff
@@ -389,6 +389,13 @@ def Scm(spec, env, overrides, recipeSet):
         except schema.SchemaError as e:
             raise ParseError("Error validating SCM after applying scmOverrides: {}".format(str(e)))
 
+    # apply scmDefaults
+    defaults = recipeSet.scmDefaults().get(spec['scm'])
+    if defaults:
+        for d in defaults:
+            if not spec.get(d):
+                spec[d] = defaults[d]
```

Comment:
I think this is equivalent to:

```suggestion
    for k, v in recipeSet.scmDefaults().get(spec['scm'], {}).items():
        spec.setdefault(k, v)
```

I guess we'll also need some test that uses the feature.

Reply:
It's always amazing ... :+1:

Reply:
:grinning: